File: csharp/Creational/01-FactoryMethod/Concrete Products/PlumbingJobCard .cs
public class PlumbingJobCard : IJobCard
{
    public string JobId    { get; } = Guid.NewGuid().ToString()[..8];
    public string Title    { get; }
    public string Category => "Plumbing";
    public string Priority { get; }

    public PlumbingJobCard(string title, string priority)
    {
        Title    = title;
        Priority = priority;
    }

    public void PrintSummary() =>
        Console.WriteLine(
            $"[{Category}] #{JobId} — {Title} (Priority: {Priority})" +
            $"\n  → Requires: Licensed plumber + water shutoff check");
}

File: typescript/Creational/01-FactoryMethod/Concrete Products/PlumbingJobCard.ts
class PlumbingJobCard implements JobCard {
  readonly jobId    = shortId();
  readonly category = "Plumbing";

  constructor(
    readonly title:    string,
    readonly priority: string
  ) {}

  printSummary(): void {
    console.log(`[${this.category}] #${this.jobId} — ${this.title} (Priority: ${this.priority})`);
    console.log("  → Requires: Licensed plumber + water shutoff check");
  }
}